src/XRP/network-clients/web-socket-network-client.ts

```diff
@@ -0,0 +1,155 @@
+import WebSocket = require('ws')
+import { XrpError, XrpErrorType } from '../shared'
+import {
+  WebSocketRequestOptions,
+  WebSocketResponse,
+  WebSocketStatusResponse,
+  WebSocketTransactionResponse,
+} from '../shared/rippled-web-socket-schema'
+
+const sleep = (ms: number): Promise<void> => {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+/**
+ * A network client for interacting with the rippled WebSocket API.
+ * @see https://xrpl.org/rippled-api.html
+ */
+export default class WebSocketNetworkClient {
+  private readonly socket: WebSocket
+  private accountCallbacks: Map<
+    string,
+    (data: WebSocketTransactionResponse) => void
+  >
+  private messageCallbacks: Map<string, (data: WebSocketResponse) => void>
```

Comment:
does it work to do ```Map<[dataType: string], (data: WebSocketResponse) => void>```?

not sure if it works in generics

Reply:
Nope, it gets mad at me